crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -400,6 +400,36 @@ impl BlobVec {
         std::slice::from_raw_parts(self.data.as_ptr() as *const UnsafeCell<T>, self.len)
     }
 
+    /// Shrinks the backing storage for the [`BlobVec`] such that the `len == capacity`.
+    ///
+    /// This runs in `O(n)` time and may require reallocating backing buffer.
+    pub fn shrink_to_fit(&mut self) {
```

Comment:
Could you add a test to this method, since everything else seems to depend on this? Some good things to test:

- Popping items from a `BlobVec` then shrinking
- Shrinking a `BlobVec` with a capacity of 0
- Shrinking a `BlobVec` with a capacity >= 1 but a length of 0